### PR TITLE
fix: display archived column for runs and searches

### PR DESCRIPTION
**Pull request by ashtonG**
## Ticket
ET-304

## Description
This adds the archived column to the columns api. It also updates the flat runs view to handle showing the archived state of runs and ensures the archived definition matches the show archived behavior.

## Test Plan
When viewing the column list for the new experiment list, the searches view, and the runs view, the user should see the "archived" column in the general tab.

## Checklist

- [x] Changes have been manually QA'd
- [x] New features have been approved by the corresponding PM
- [x] User-facing API changes have the "User-facing API Change" label
- [x] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by AmanuelAaron — APPROVED** [review]
Backend LGTM

File: master/internal/api_project.go
Comment (EmilyBonar) [inline master/internal/api_project.go:181]: q: would it be possible to add a boolean type in this pr, or is that out of scope?

File: webui/react/src/pages/FlatRuns/columns.ts
Comment (EmilyBonar) [inline webui/react/src/pages/FlatRuns/columns.ts:64]: nit: might be worth explicitly filtering out "archived" to avoid issues if the array gets reordered

**Review by EmilyBonar — APPROVED** [review]
Web side of things looks good. Backend seems fine but wait for their approval.

File: master/internal/api_project.go
Comment (ashtonG) [reply master/internal/api_project.go:181]: Let's handle it separately because `parentArchived` in runs + metadata is also affected by that change.

**Pull request merged**
